components/pluralrules/src/rules/mod.rs

```diff
@@ -0,0 +1,146 @@
+//!                 modulus: None,
+//!             },
+//!             operator: Operator::Eq,
+//!             range_list: RangeList(Box::new([
+//!                 RangeListItem::Value(
+//!                     Value(1)
+//!                 )
+//!             ]))
+//!         },
+//!         Relation {
+//!             expression: Expression {
+//!                 operand: Operand::V,
+//!                 modulus: None,
+//!             },
+//!             operator: Operator::Eq,
+//!             range_list: RangeList(Box::new([
+//!                 RangeListItem::Value(
+//!                     Value(0)
+//!                 )
+//!             ]))
+//!         },
+//!     ])),
+//! ])));
+//! ```
+//!
+//! Finally, we can pass this [`AST`] (in fact, just the [`Condition`] node),
+//! to a resolver alongside the [`PluralOperands`] to test if the Rule
+//! matches:
+//!
+//! ```
+//! use icu_pluralrules::rules::{test_condition, parse};
+//! use icu_pluralrules::PluralOperands;
+//!
+//! let input = "i = 1 and v = 0 @integer 1";
+//!
+//! let operands = PluralOperands::from(1_u32);
+//!
+//! let ast = parse(input.as_bytes())
+//!     .expect("Parsing failed.");
+//!
+//! assert!(test_condition(&ast, &operands));
+//! ```
+//!
+//! Since the rule for [`PluralCategory::One`] matches, we will return this category.
+//! Otherwise, we'd test the next rule, in this case [`PluralCategory::Other`], which has an
+//! empty [`Condition`], meaning that it'll match all operands.
+//!
+//! # Summary
+//!
+//! For [`PluralRuleType::Cardinal`] in English, we can summarize the logic as:
+//!
+//! If [`PluralOperands::i`] is `1` and [`PluralOperands::v`] is `0`, [`PluralCategory::One`]
+//! should be used, otherwise [`PluralCategory::Other`] should be used.
+//!
+//! For other locales, there are more [`PluralCategories`] and more complicated [`Rules`].
+//!
+//! # Difference between Category and Number
+//!
+//! While in English [`PluralCategory::One`] overlaps with an integer value `1`,
+//! in other languages, the category may be used for other numbers as well.
+//!
+//! For example, in Russian [`PluralCategory::One`] matches numbers such as `11`, `21`, `121` etc.
+//!
+//! [`PluralCategory`]: ../enum.PluralCategory.html
+//! [`PluralCategories`]: ../enum.PluralCategory.html
+//! [`PluralCategory::One`]: ../enum.PluralCategory.html#variant.One
+//! [`PluralCategory::Other`]: ../enum.PluralCategory.html#variant.Other
+//! [`PluralOperands`]: ../struct.PluralOperands.html
+//! [`PluralOperands::i`]: ../struct.PluralOperands.html#structfield.i
+//! [`PluralOperands::v`]: ../struct.PluralOperands.html#structfield.v
+//! [`PluralRuleType::Cardinal`]: ../enum.PluralRuleType.html#variant.Cardinal
+//! [`Rule`]: ../rules/ast/struct.Rule.html
+//! [`Rules`]: ../rules/ast/struct.Rule.html
+//! [`Condition`]: ../rules/ast/struct.Condition.html
+//! [`Sample`]: ../rules/ast/struct.Sample.html
+//! [`AST`]: ../rules/ast/index.html
+pub mod ast;
```

Comment:
Why are you making the stuff in this module public?

Reply:
Partially, to allow benchmarks on parser, but also to allow people to use the Parser/AST/resolver on their own. The simplest use cases may be helper macro crates that will take a string and parse it into an AST that will be stored in some serialized form.

I could remove the public access for now, and wait till we have a use for the parser/ast/resolver but would have to figure out how to make it work for benchmarks and tests.